src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/disk_encryption.py

```diff
@@ -102,30 +125,40 @@ def encrypt_vm(cmd, resource_group_name, vm_name,  # pylint: disable=too-many-lo
     # 2. we are ready to provision/update the disk encryption extensions
     # The following logic was mostly ported from xplat-cli
     public_config = {
-        'AADClientID': aad_client_id,
-        'AADClientCertThumbprint': aad_client_cert_thumbprint,
         'KeyVaultURL': disk_encryption_keyvault_url,
         'VolumeType': volume_type,
         'EncryptionOperation': 'EnableEncryption' if not encrypt_format_all else 'EnableEncryptionFormatAll',
         'KeyEncryptionKeyURL': key_encryption_key,
         'KeyEncryptionAlgorithm': key_encryption_algorithm,
         'SequenceVersion': sequence_version,
     }
-    private_config = {
+    if use_new_ade:
+        public_config.update({
+            "KeyVaultResourceId": disk_encryption_keyvault,
+            "KekVaultResourceId": key_encryption_keyvault if key_encryption_key else '',
```

Comment:
Typo 'KekVaultResourceId' I assume

Reply:
This time, it is a not a typo. `kek` stands for `key-vault-key` for securer encryption